hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCreateRequest.java

```diff
@@ -236,19 +235,19 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
           omMetadataManager.getBucketKey(volumeName, bucketName));
 
       // If FILE_EXISTS we just override like how we used to do for Key Create.
-      List< OzoneAcl > inheritAcls;
       if (LOG.isDebugEnabled()) {
         LOG.debug("BucketName: {}, BucketLayout: {}",
             bucketInfo.getBucketName(), bucketInfo.getBucketLayout());
       }
+
+      OMFileRequest.OMPathInfo pathInfo =
```

Comment:
This needs to be done only if its EnableFileSystemPaths for Legacy, otherwise needs to be null.

Reply:
> This needs to be done only if its EnableFileSystemPaths for Legacy, otherwise needs to be null.

@sumitagrawl Thanks for the advice！Yes, you are right. But I found that even if `EnableFileSystemPaths` is `false`, the pathInfo can still be fetched before the judgment. Subsequent logic is still compatible. Do you think so?
I added `TestOMKeyCreateRequest` to verify it.

Reply:
@whbing For OBS case, there is no directory or parent path, all key belongs to bucket only. If above is used to get pathInfo, and if mistakely key name pattern follows FSO naming, and parent directory exist, then it will inherit the ACL of the parent key present (which is another Key).
So this can be incorrect to have ACL from another key in OBS case.

So we need do only inside if EnableFileSystem, and null can be passed in prepareKeyInfo for pathInfo.

Reply:
> @whbing For OBS case, there is no directory or parent path, all key belongs to bucket only. If above is used to get pathInfo, and if mistakely key name pattern follows FSO naming, and parent directory exist, then it will inherit the ACL of the parent key present (which is another Key). So this can be incorrect to have ACL from another key in OBS case.
> 
> So we need do only inside if EnableFileSystem, and null can be passed in prepareKeyInfo for pathInfo.

@sumitagrawl  Thanks, got it and already fixed.